src/bugsplat/bugsplat.py

```diff
@@ -1,100 +1,120 @@
 import json
-import os
-import requests
 import traceback
-import sys
+from logging import Logger, getLogger
+from os import PathLike
+from pathlib import PurePath
+from typing import List
+
+import requests
+
 
 class BugSplat:
-    def __init__(self, database, application, version):
+    def __init__(self,
+                 database: str,
+                 application: str,
+                 version: str,
+                 logger: Logger = None):
         self.database = database
         self.application = application
         self.version = version
-        self.additionalFilePaths = []
-        self.appKey = ''
+        self.additional_file_paths = []
+        self.app_key = ''
         self.description = ''
         self.email = ''
         self.user = ''
+        self.logger = logger or getLogger(__name__)
 
-    def setDefaultAdditionalFilePaths(self, additionalFilePaths):
-        self.additionalFilePaths = additionalFilePaths
+    def set_default_additional_file_paths(self, additional_file_paths: List[PathLike]):
+        self.additional_file_paths = additional_file_paths
 
-    def setDefaultAppKey(self, key):
-        self.appKey = key
+    def set_default_app_key(self, key: str):
```

Comment:
word! when did Python get types? lol

Reply:
typing module added in 3.5, so quite a while ago 😁

https://docs.python.org/3/library/typing.html